packages/kit/src/loader/config.ts

```diff
@@ -37,6 +38,18 @@ export async function loadNuxtConfig (opts: LoadNuxtConfigOptions): Promise<Nuxt
 
   nuxtConfig._nuxtConfigFile = configFile
   nuxtConfig._nuxtConfigFiles = [configFile]
+
+  // Resolve `rootDir` & `srcDir` of layers
+  for (const layer of layers) {
+    layer.config.rootDir = layer.cwd
```

Comment:
We might keep overriden `rootDir` value by `layer.config` if it exists:

```suggestion
    layer.config.rootDir = resolve(layer.cwd, layer.config.rootDir)
```